saleor/graphql/checkout/tests/mutations/test_checkout_create.py

```diff
@@ -2910,3 +2910,211 @@ def test_checkout_create_no_billing_address_providing_save_address_raising_error
     error = errors[0]
     assert error["field"] == "saveBillingAddress"
     assert error["code"] == CheckoutErrorCode.MISSING_ADDRESS_DATA.name
+
+
+def test_checkout_create_with_metadata_and_private_metadata(
+    staff_api_client, graphql_address_data, channel_USD, permission_manage_checkouts
+):
+    # Given
+    test_email = "test@example.com"
+
+    metadata_key = "metadata_key"
+    metadata_key_2 = "metadata_key_2"
+    metadata_value = "metadata_value"
+
+    variables = {
+        "checkoutInput": {
+            "channel": channel_USD.slug,
+            "lines": [],
+            "email": test_email,
+            "metadata": [{"key": metadata_key, "value": metadata_value}],
+            # Make it more diverse, set 2 keys
+            "privateMetadata": [
+                {"key": metadata_key, "value": metadata_value},
+                {"key": metadata_key_2, "value": metadata_value},
+            ],
+        }
+    }
+
+    assert not Checkout.objects.exists()
```

Comment:
I'm unsure if we should check if the database is empty in each test. IMO empty database is a default state that we could assume as a starting point for each test.

Reply:
For me it's fine because it additionally ensures that our initial state is valid

But it's out of my scope- every test in this file does it, so probably it's separate discussion either we need this or should remove it

Reply:
Fully agreed. I just thinking about the test performance problem. Each optional database operation in the test increases execution time. We should clear those tests in separate PR.